### ISSUE TITLE
[BUG] Firebase Init Failure for Org Repository Hosting even with Organizational Access Allowed

### ISSUE THREAD

**Issue by joshelson** [FirebaseExtended/action-hosting-deploy#136]
Found the other open issue with organizational access not allowed, but I have organizational access to Firebase CLI and I am still getting this error.

Debug output here:

```
 ✔  Success! Logged into GitHub as joshelson

[2021-07-06T00:55:22.569Z] >>> [apiv2][query] GET https://api.github.com/repos/undefined [none]
[2021-07-06T00:55:24.469Z] <<< [apiv2][status] GET https://api.github.com/repos/undefined 404
[2021-07-06T00:55:24.469Z] <<< [apiv2][body] GET https://api.github.com/repos/undefined {"message":"Not Found","documentation_url":"https://docs.github.com/rest"}

Error: HTTP Error: 404, Not Found
[2021-07-06T00:55:26.197Z] Error Context: {
  "body": {
    "message": "Not Found",
    "documentation_url": "https://docs.github.com/rest",
    "error": {
      "message": "Not Found"
    }
  },
  "response": {
    "statusCode": 404
  }
}
```

**Comment by riker09** [comment]
You mean issue #40? Did you try the suggested solution from [this comment](https://github.com/FirebaseExtended/action-hosting-deploy/issues/40#issuecomment-748017911)? My apologies, but it isn't clear from your comment.

**Comment by riker09** [comment]
@3ximus Your error seems to be something else: `TypeError: Cannot read property 'predeploy' of undefined` I *think* this might come from a misconfigured `firebase.json` and more specifically the `functions` configuration.

Here's an excerpt from my `firebase.json`:

```json
{
  "functions": {
    "source": "functions",
    "predeploy": [
      "npm run lint --workspace \"$RESOURCE_DIR\"",
      "npm run build --workspace \"$RESOURCE_DIR\""
    ]
  },

  // ...
}
```

tl;dr: Check and make sure that your `firebase.json` is valid.

**Comment by 3ximus** [comment]
I generated it through the CLI but selecting both these options for Hosting at the same time created this problem for me: 
```
Hosting: Configure files for Firebase Hosting and (optionally) set up GitHub Action deploys
Hosting: Set up GitHub Action deploys
```
Choosing only the first option fixes it for me.
Cheers
